src/Slices/Dashboard/UI/helper.ts

```diff
@@ -61,12 +61,79 @@ export const formatMetricsToStacked = (
           max = tempMax;
         }
       });
+      tempCharState = tempCharState.map((metric) => formatValues(metric));
     }
   } else {
-    tempCharState = [metrics as Metric];
+    tempCharState = [
+      metrics.name.includes("service_count")
+        ? formatValues(metrics as Metric)
+        : (metrics as Metric),
+    ];
     max = (metrics as Metric).data
       .flatMap((value) => (value !== null ? value : 0))
       .sort((a, b) => a - b)[(metrics as Metric).data.flat().length - 1];
   }
   return [tempCharState, max];
 };
+
+const formatValues = (metrics: Metric) => {
+  const newMetrics = metrics.data.map((data, index) => {
+    if (data == null || index === 0) {
+      return null;
+    }
+    if (metrics.data[index - 1] !== null) {
+      if (data > (metrics.data[index - 1] as number)) {
+        return Math.floor(data);
+      } else {
+        return Math.ceil(data);
+      }
+    }
+    return Math.round(data);
+  });
+  return {
+    ...metrics,
+    data: newMetrics,
+  };
+};
+
+/**
+ * Replace null values between two numbers(no matter how many nulls they are in between,
+ * as long as there are some boundaries) with interpolation on line between those two numbers.
+ *
+ * @param metrics
+ * @returns metrics with interpolated values instead null values
+ */
+export const interpolateMetrics = (metrics: (number | null)[]) => {
+  const newMetric: (number | null)[] = [];
+  let nextNumber = 0;
+
+  metrics.forEach((value, index) => {
+    if (nextNumber === -1) {
+      newMetric.push(null);
+    }
+    if (value === null && (index === 0 || index === metrics.length - 1)) {
+      newMetric.push(null);
+    } else if (value === null && newMetric[index - 1] === null) {
+      newMetric.push(null);
+    } else if (value === null && newMetric[index - 1] !== null) {
+      if (metrics[index + 1] !== null) {
+        newMetric.push(lerp(newMetric[index - 1], metrics[index + 1], 0.5));
```

Comment:
why 0.5 ?

Reply:
0.5 is used as we want the middle value(average) between these two values, but I now removed that if and moved all of the interpolation inside last else if